app/scripts/components/analysis/results/timeseries-data.ts

```diff
@@ -155,36 +159,32 @@ async function getDatasetAssets(
           include: ['assets.cog_default.href', 'properties.start_datetime'],
           exclude: ['collection', 'links']
         },
-        // TODO: Only supports intersection on a single geometry???
-        intersects: aoi.geometry,
         filter: {
           op: 'and',
           args: [
             {
-              op: '>=',
+              op: 'eq',
               args: [
                 {
-                  property: 'datetime'
+                  property: 'collection'
                 },
-                date.start
+                id
               ]
             },
             {
-              op: '<=',
-              args: [
-                {
-                  property: 'datetime'
-                },
-                date.end
-              ]
+              op: 's_intersects',
+              args: [{ property: 'geometry' }, aoi.geometry]
             },
             {
-              op: 'eq',
+              op: 't_intersects',
               args: [
+                { property: 'datetime' },
                 {
-                  property: 'collection'
-                },
-                id
+                  interval: [
+                    date.start,
+                    date.end
+                  ]
+                }
```

Comment:
See #307 for explanation.

Reply:
Too bad we are duplicating logic here. Is there a way we could reuse `analysis/utils/getFilterPayload` here (sending a single id as the `collections` arg).

Otherwise great use case for https://github.com/developmentseed/ds-business/issues/1310

Reply:
@nerik Indeed we can! done! :)

Reply:
@danielfdsilva Perfect!